backend-server/src/main/java/org/dice/ida/action/def/VisualizeAction.java

```diff
@@ -417,19 +416,27 @@ private void processBinsForNumericLabels(int binSize, String xAxisColumn, String
 		double max = values.get(values.size() - 1);
 		double binVal;
 		double intervalBegin;
+		String valueString;
 		for (double i = min; i < max; i += binSize) {
 			graphItems.put(i + " - " + (i + binSize - 1), 0.0);
 			labelCounts.put(i + " - " + (i + binSize - 1), 1);
 		}
 		for (Map<String, String> entry : tableData) {
 			try {
-				binVal = Double.parseDouble(entry.get(xAxisColumn));
-				intervalBegin = binVal - (binVal % binSize);
-				xValue = intervalBegin + " - " + (intervalBegin + binSize - 1);
+				valueString = entry.get(xAxisColumn);
+				if (TextUtil.isDoubleString(valueString)) {
+					binVal = Double.parseDouble(valueString);
+					intervalBegin = binVal - (binVal % binSize);
+					xValue = intervalBegin + " - " + (intervalBegin + binSize - 1);
+					updateGraphItemList(xValue, entry.get(yAxisColumn), yAxisColumnType, labelCounts);
+				} else if (valueString.equalsIgnoreCase(IDAConst.NULL_VALUE_IDENTIFIER)) {
+					xValue = entry.get(xAxisColumn);
+					updateGraphItemList(xValue, entry.get(yAxisColumn), yAxisColumnType, labelCounts);
+				}
 			} catch (NumberFormatException ex) {
-				xValue = entry.get(xAxisColumn);
+				ex.printStackTrace();
```

Comment:
Exception handling should not be done like this. Please include proper logic and logs.